### PR TITLE
Update Proguard rules on README.md

### LINKED ISSUE #3371: Proguard rules on README.md not working
**Current README.md rules:**
https://github.com/square/okhttp/commit/f87d32984dd9b1f5a7092b52dbecbafd784bd764
```
-dontwarn okio.**
-dontwarn javax.annotation.Nullable
-dontwarn javax.annotation.ParametersAreNonnullByDefault
```

**Error:**
```
Warning: okhttp3.internal.huc.DelegatingHttpsURLConnection: can't find referenced class org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
Warning: okhttp3.internal.huc.DelegatingHttpsURLConnection: can't find referenced class org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
Warning: okhttp3.internal.huc.DelegatingHttpsURLConnection: can't find referenced class org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
Warning: okhttp3.internal.huc.OkHttpsURLConnection: can't find referenced class org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
Warning: okhttp3.internal.huc.OkHttpsURLConnection: can't find referenced class org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
Warning: okhttp3.internal.huc.OkHttpsURLConnection: can't find referenced class org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
```

We need to add `-dontwarn okhttp3.**`.

### PR DESCRIPTION
**Pull request by jaredsburrows**
@swankjesse Closes #3371

### CODE REVIEW

**Comment by jaredsburrows** [comment]
@swankjesse Should we also add to the Github pages website as well? Similar to http://square.github.io/retrofit/?

File: README.md [hunk 1]
```diff
@@ -50,10 +50,9 @@ ProGuard
 
 If you are using ProGuard you might need to add the following options:
 ```
+-dontwarn okhttp3.**
 -dontwarn okio.**
--dontwarn javax.annotation.Nullable
--dontwarn javax.annotation.ParametersAreNonnullByDefault
-```
+-dontwarn javax.annotation.**```
```
Comment (PromanSEW) [inline README.md]: Backticks should be on the new line

**Comment by jaredsburrows** [comment]
Yes

File: README.md (hunk 1 above)
Comment (jaredsburrows) [reply README.md]: Fixed. Thanks.

**Pull request merged**